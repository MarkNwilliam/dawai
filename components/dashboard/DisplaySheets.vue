<template>
  <v-layout row wrap>
    <v-flex xs12 sm4>
      <v-sheet
        min-height="265"
        max-height="335"
        class="mx-5 mt-5 mb-12 pa-5 main-gradient"
        dark
      >
        <v-row justify="center" align="center">
          <div class="text-center ma-5">
            <v-progress-circular
              rotate="90"
              size="128"
              value="72"
              width="8"
              color="white"
              >72%</v-progress-circular
            >
          </div>
        </v-row>
        <v-row align="center" justify="center">
          <span class="headline text-center">Yearly Income Goal</span>
        </v-row>
      </v-sheet>
    </v-flex>
    <v-flex xs12 sm4>
      <v-sheet
        min-height="265"
        max-height="335"
        class="mx-5 mt-5 mb-12 pa-5 main-gradient"
        dark
      >
        <v-row class="mb-9 mt-2" justify="center" align="center">
          <v-icon color="white" size="112">{{ mdiCashMultiple }}</v-icon>
        </v-row>
        <v-row align="center" justify="center">
          <span class="headline text-center">55,700.00 <sup>SDG</sup></span>
        </v-row>
        <v-row align="center" justify="center">
          <span class="text-center">Avg. Monthly Income</span>
        </v-row>
      </v-sheet>
    </v-flex>
    <v-flex xs12 sm4>
      <v-sheet
        min-height="265"
        max-height="335"
        class="mx-5 mt-5 mb-12 pa-5 main-gradient"
        dark
      >
        <v-row justify="center" align="center">
          <div class="text-center ma-5">
            <v-progress-circular
              rotate="90"
              size="128"
              value="28"
              width="8"
              color="white"
              >28%</v-progress-circular
            >
          </div>
        </v-row>
        <v-row align="center" justify="center">
          <span class="headline text-center">Monthly Income Goal</span>
        </v-row>
      </v-sheet>
    </v-flex>
  </v-layout>
</template>

<script>
import { mdiCashMultiple } from '@mdi/js'
export default {
  data: () => ({
    mdiCashMultiple
  })
}
</script>
