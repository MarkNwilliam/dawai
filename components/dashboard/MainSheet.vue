<template>
  <div class="row">
    <v-sheet
      class="text-center my-10 pa-5 heavy-br"
      color="primary"
      width="100%"
      elevation="6"
      dark
    >
      <span class="font-weight-thin display-2">Dashboard</span>
    </v-sheet>
  </div>
</template>
