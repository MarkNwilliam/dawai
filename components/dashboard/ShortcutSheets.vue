<template>
  <v-row align="center" justify="center">
    <v-col cols="12" sm="6" md="3" lg="3">
      <v-sheet class="transparent" dark>
        <v-row align="center" justify="center">
          <v-card color="transparent" flat>
            <v-icon color="primary" size="160">{{ mdiReceipt }}</v-icon>
            <v-card-actions class="text-center">
              <v-btn to="/finance" block class="black--text" color="white"
                >Create Invoice</v-btn
              >
            </v-card-actions>
          </v-card></v-row
        ></v-sheet
      >
    </v-col>
    <v-col cols="12" sm="6" md="3" lg="3">
      <v-sheet class="transparent" dark>
        <v-row align="center" justify="center">
          <v-card color="transparent" flat>
            <v-icon color="primary" size="160">{{ mdiBottleTonicPlus }}</v-icon>
            <v-card-actions class="text-center">
              <v-btn to="/medicine" block class="black--text" color="white"
                >Add Medicine</v-btn
              >
            </v-card-actions>
          </v-card></v-row
        ></v-sheet
      >
    </v-col>
    <v-col cols="12" sm="6" md="3" lg="3">
      <v-sheet class="transparent" dark>
        <v-row align="center" justify="center">
          <v-card color="transparent" flat>
            <v-icon color="primary" size="160">{{ mdiBell }}</v-icon>
            <v-card-actions class="text-center">
              <v-btn class="black--text" block color="white"
                >0 Notifications</v-btn
              >
            </v-card-actions>
          </v-card></v-row
        ></v-sheet
      >
    </v-col>
  </v-row>
</template>

<script>
import {
  mdiReceipt,
  mdiBottleTonicPlus,
  mdiFileChartOutline,
  mdiBell
} from '@mdi/js'
export default {
  data: () => ({
    mdiReceipt,
    mdiBottleTonicPlus,
    mdiFileChartOutline,
    mdiBell
  })
}
</script>
