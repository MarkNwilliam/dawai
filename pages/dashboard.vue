<template>
  <v-layout row wrap>
    <v-flex xs12>
      <MainSheet />
    </v-flex>
    <v-flex xs12>
      <DisplaySheets />
    </v-flex>
    <v-flex xs12 sm6 md4>
      <v-card flat>
        <LineChart />
      </v-card>
    </v-flex>
    <v-flex xs12 sm6 md4>
      <v-card flat>
        <PieChart />
      </v-card>
    </v-flex>
    <v-flex xs12 sm6 md4>
      <v-card flat>
        <HorizontalBarChart />
      </v-card>
    </v-flex>
    <v-flex class="my-5" xs12>
      <v-card flat>
        <TimeChart />
      </v-card>
    </v-flex>
  </v-layout>
</template>

<script>
import MainSheet from '@/components/dashboard/MainSheet'
import LineChart from '@/components/dashboard/charts/LineChart'
import PieChart from '@/components/dashboard/charts/PieChart'
import HorizontalBarChart from '@/components/dashboard/charts/HorizontalBarChart'
import TimeChart from '@/components/dashboard/charts/TimeChart'
import DisplaySheets from '@/components/dashboard/DisplaySheets'
export default {
  layout: ({ isMobile }) => (isMobile ? 'mobile' : 'default'),
  components: {
    MainSheet,
    LineChart,
    PieChart,
    HorizontalBarChart,
    TimeChart,
    DisplaySheets
  },
  head: () => ({
    title: 'Dashboard'
  })
}
</script>
