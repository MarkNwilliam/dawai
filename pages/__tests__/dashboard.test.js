// Libraries
import Vuetify from 'vuetify'

// Component
import { mount, createLocalVue } from '@vue/test-utils'
import dashboard from '../dashboard'

// Utilities
const localVue = createLocalVue()

describe('dashboard.vue', () => {
  let vuetify

  beforeEach(() => {
    vuetify = new Vuetify()
  })

  it('should render', (done) => {
    const wrapper = mount(dashboard, {
      localVue,
      vuetify
    })

    // With jest we can create snapshot files of the HTML output
    expect(wrapper.html()).toMatchSnapshot()

    done()
  })
})
