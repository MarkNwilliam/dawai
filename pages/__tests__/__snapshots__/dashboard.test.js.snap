// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dashboard.vue should render 1`] = `"<v-row align=\\"center\\" justify=\\"center\\" dense=\\"\\"><v-col cols=\\"12\\" sm=\\"12\\" md=\\"12\\" lg=\\"12\\" xl=\\"12\\"><div class=\\"row\\"><v-sheet color=\\"primary\\" width=\\"100%\\" elevation=\\"6\\" dark=\\"\\" class=\\"text-center my-10 pa-5 heavy-br\\"><span class=\\"font-weight-thin display-2\\">Dashboard</span></v-sheet></div></v-col> <v-col cols=\\"12\\" sm=\\"12\\" md=\\"12\\" lg=\\"12\\" xl=\\"12\\"><v-row align=\\"center\\" justify=\\"center\\"><v-col cols=\\"12\\" sm=\\"6\\" md=\\"3\\" lg=\\"3\\"><v-sheet dark=\\"\\" class=\\"transparent\\"><v-row align=\\"center\\" justify=\\"center\\"><v-card color=\\"transparent\\" flat=\\"\\"><v-icon color=\\"primary\\" size=\\"160\\">M3,22L4.5,20.5L6,22L7.5,20.5L9,22L10.5,20.5L12,22L13.5,20.5L15,22L16.5,20.5L18,22L19.5,20.5L21,22V2L19.5,3.5L18,2L16.5,3.5L15,2L13.5,3.5L12,2L10.5,3.5L9,2L7.5,3.5L6,2L4.5,3.5L3,2M18,9H6V7H18M18,13H6V11H18M18,17H6V15H18V17Z</v-icon> <v-card-actions class=\\"text-center\\"><v-btn to=\\"/finance\\" block=\\"\\" color=\\"white\\" class=\\"black--text\\">Create Invoice</v-btn></v-card-actions></v-card></v-row></v-sheet></v-col> <v-col cols=\\"12\\" sm=\\"6\\" md=\\"3\\" lg=\\"3\\"><v-sheet dark=\\"\\" class=\\"transparent\\"><v-row align=\\"center\\" justify=\\"center\\"><v-card color=\\"transparent\\" flat=\\"\\"><v-icon color=\\"primary\\" size=\\"160\\">M13 4H11L10 2H14L13 4M14 8V6H15V5H9V6H10V8C7.24 8 5 10.24 5 13V22H19V13C19 10.24 16.76 8 14 8M16 17H13V20H11V17H8V15H11V12H13V15H16V17Z</v-icon> <v-card-actions class=\\"text-center\\"><v-btn to=\\"/medicine\\" block=\\"\\" color=\\"white\\" class=\\"black--text\\">Add Medicine</v-btn></v-card-actions></v-card></v-row></v-sheet></v-col> <v-col cols=\\"12\\" sm=\\"6\\" md=\\"3\\" lg=\\"3\\"><v-sheet dark=\\"\\" class=\\"transparent\\"><v-row align=\\"center\\" justify=\\"center\\"><v-card color=\\"transparent\\" flat=\\"\\"><v-icon color=\\"primary\\" size=\\"160\\">M21,19V20H3V19L5,17V11C5,7.9 7.03,5.17 10,4.29C10,4.19 10,4.1 10,4A2,2 0 0,1 12,2A2,2 0 0,1 14,4C14,4.1 14,4.19 14,4.29C16.97,5.17 19,7.9 19,11V17L21,19M14,21A2,2 0 0,1 12,23A2,2 0 0,1 10,21</v-icon> <v-card-actions class=\\"text-center\\"><v-btn block=\\"\\" color=\\"white\\" class=\\"black--text\\">0 Notifications</v-btn></v-card-actions></v-card></v-row></v-sheet></v-col></v-row></v-col> <v-col cols=\\"12\\" sm=\\"12\\" md=\\"12\\" lg=\\"12\\" xl=\\"12\\"><v-row align=\\"center\\" justify=\\"center\\" dense=\\"\\"><v-col cols=\\"12\\" sm=\\"12\\" md=\\"6\\" lg=\\"4\\" xl=\\"4\\"><v-sheet min-height=\\"265\\" max-height=\\"335\\" dark=\\"\\" class=\\"mx-5 mt-5 mb-12 pa-5 transparent\\"><v-row justify=\\"center\\" align=\\"center\\"><div class=\\"text-center ma-5\\"><v-progress-circular rotate=\\"90\\" size=\\"128\\" value=\\"72\\" width=\\"8\\" color=\\"primary\\">72%</v-progress-circular></div></v-row> <v-row align=\\"center\\" justify=\\"center\\"><span class=\\"headline text-center black--text\\">Yearly Income Goal</span></v-row></v-sheet></v-col> <v-col cols=\\"12\\" sm=\\"12\\" md=\\"6\\" lg=\\"4\\" xl=\\"4\\"><v-sheet min-height=\\"265\\" max-height=\\"335\\" dark=\\"\\" class=\\"mx-5 mt-5 mb-12 pa-5 transparent\\"><v-row justify=\\"center\\" align=\\"center\\" class=\\"mb-9 mt-2\\"><v-icon color=\\"primary\\" size=\\"112\\">M5,6H23V18H5V6M14,9A3,3 0 0,1 17,12A3,3 0 0,1 14,15A3,3 0 0,1 11,12A3,3 0 0,1 14,9M9,8A2,2 0 0,1 7,10V14A2,2 0 0,1 9,16H19A2,2 0 0,1 21,14V10A2,2 0 0,1 19,8H9M1,10H3V20H19V22H1V10Z</v-icon></v-row> <v-row align=\\"center\\" justify=\\"center\\"><span class=\\"headline text-center black--text\\">55,700.00 <sup>SDG</sup></span></v-row> <v-row align=\\"center\\" justify=\\"center\\"><span class=\\"text-center black--text\\">Avg. Monthly Income</span></v-row></v-sheet></v-col> <v-col cols=\\"12\\" sm=\\"12\\" md=\\"6\\" lg=\\"4\\" xl=\\"4\\"><v-sheet min-height=\\"265\\" max-height=\\"335\\" dark=\\"\\" class=\\"mx-5 mt-5 mb-12 pa-5 transparent\\"><v-row justify=\\"center\\" align=\\"center\\"><div class=\\"text-center ma-5\\"><v-progress-circular rotate=\\"90\\" size=\\"128\\" value=\\"28\\" width=\\"8\\" color=\\"primary\\">28%</v-progress-circular></div></v-row> <v-row align=\\"center\\" justify=\\"center\\"><span class=\\"headline text-center black--text\\">Monthly Income Goal</span></v-row></v-sheet></v-col></v-row></v-col> <v-col cols=\\"12\\" sm=\\"6\\" md=\\"4\\" lg=\\"4\\" xl=\\"4\\"><div class=\\"\\"><canvas id=\\"line-chart\\" width=\\"0\\" height=\\"0\\" style=\\"display: block;\\" class=\\"chartjs-render-monitor\\"></canvas></div></v-col> <v-col cols=\\"12\\" sm=\\"6\\" md=\\"4\\" lg=\\"4\\" xl=\\"4\\"><div class=\\"\\"><canvas id=\\"pie-chart\\" width=\\"0\\" height=\\"0\\" style=\\"display: block;\\" class=\\"chartjs-render-monitor\\"></canvas></div></v-col> <v-col cols=\\"12\\" sm=\\"6\\" md=\\"4\\" lg=\\"4\\" xl=\\"4\\"><div class=\\"\\"><canvas id=\\"horizontalbar-chart\\" width=\\"0\\" height=\\"0\\" style=\\"display: block;\\" class=\\"chartjs-render-monitor\\"></canvas></div></v-col> <v-col cols=\\"12\\" sm=\\"12\\" md=\\"12\\" lg=\\"12\\" xl=\\"12\\"><div class=\\"\\"><canvas id=\\"line-chart\\" width=\\"0\\" height=\\"0\\" style=\\"display: block;\\" class=\\"chartjs-render-monitor\\"></canvas></div></v-col></v-row>"`;
